target/linux/generic/hack-5.10/230-openwrt_lzma_options.patch

```diff
@@ -13,11 +13,12 @@ Signed-off-by: Imre Kaloz <kaloz@openwrt.org>
 
 --- a/lib/decompress.c
 +++ b/lib/decompress.c
-@@ -53,6 +53,7 @@ static const struct compress_format comp
+@@ -53,6 +53,8 @@ static const struct compress_format comp
  	{ {0x1f, 0x9e}, "gzip", gunzip },
  	{ {0x42, 0x5a}, "bzip2", bunzip2 },
  	{ {0x5d, 0x00}, "lzma", unlzma },
 +	{ {0x6d, 0x00}, "lzma-openwrt", unlzma },
++	{ {0x62, 0x00}, "lzma-openwrt", unlzma },
```

Comment:
hmm, LZMA streams do not start with a "magic" string/byte. Instead the first byte is used to encode the LZMA compression settings for the decompressor (i.e. context bits, position bits, etc). So doing the compression algorithm detection on that will lead to issues. 

From what I can tell, this special 0x62 is caused by the x86-target patch:
https://github.com/openwrt/openwrt/blob/master/target/linux/x86/patches-5.10/011-tune_lzma_options.patch

`-   cmd_lzma = { cat $(real-prereqs) | $(LZMA) e -d20 -lc1 -lp2 -pb2 -eos -si -so; $(size_append); } > $@`
`+      cmd_lzma = { cat $(real-prereqs) | $(LZMA) e -lc8 -eos -si -so; $(size_append); } > $@`

that ( -lc8 is what's causing it). So alternatively, that hunk (or maybe the whole 011-tune_lzma_options.patch, if it has become unneccessary could be removed to fix this as well).

Reply:
... @ptpt52, any updates there?

yes: this is a problem and it should be fixed. I would prefer that it does not involve adding to the hacks though.

Cheers
Christian